### ISSUE TITLE
"pulumi preview --diff" doesn't respect "additionalSecretsOutput" property settings.

### ISSUE THREAD

**Issue by mark-bixler** [pulumi/pulumi#4058]
When running a `pulumi preview --diff` the output continues to show a password, even though the property `additionalSecretsOuptut` has been set for that value.

Example Code:

```
this._directoryService = new aws.directoryservice.Directory(
      `${adConfig.name}-directory`, {
      edition: 'Standard',
      password: adminPwVersion.secretString,
      name: directory.name,
      shortName: directory.shortname,
      type: 'MicrosoftAD',
      vpcSettings: {
        subnetIds: [directory.subnet1, directory.subnet2],
        vpcId: directory.vpc
        },
      tags: adConfig.tags
      },
      {
        parent: this,
        provider: acct,
        additionalSecretOutputs: ['password']
      }
    );
```

Result:
```
...
  edition: "standard
  name: "aws.activedirectory.com"
  password: "Password123"
 ...
```

**Comment by lukehoban** [comment]
Related: https://github.com/pulumi/pulumi/issues/3138

**Comment by mikhailshilkov** [comment]
Indeed, I think this is an instance of #3138 and it will resolve if we implement that enhancement. I'll go ahead and close this issue as dupe, since it's unlikely we will fix it without addressing the whole #3138.
